pkg/controller/operators/catalog/operator.go

```diff
@@ -622,6 +628,43 @@ func (o *Operator) syncConnection(logger *logrus.Entry, in *v1alpha1.CatalogSour
 	return
 }
 
+func (o *Operator) checkBackingPodStatus(logger *logrus.Entry, in *v1alpha1.CatalogSource) (out *v1alpha1.CatalogSource, continueSync bool, syncError error) {
+	if in.Spec.Address != "" {
+		// if this is an imageless source type, there's no backing pod to check
+		return in, true, nil
+	}
+
+	out = in.DeepCopy()
+
+	selector := map[string]string{reconciler.CatalogSourceLabelKey: in.GetName()}
```

Comment:
shouldn't this use the pod decorator to get this selector?

Reply:
I didn't see a way to have access to the decorator at this level, so I refactored to put some commonality into the reconciler package and use that instead.